### ISSUE TITLE
`X-Endpoint-API-UserInfo` does not include `claims`

### ISSUE THREAD

**Issue by DazWilkin** [GoogleCloudPlatform/esp-v2#544]
This isn't an issue for me but is an inconsistency.

Google's documentation [here](https://cloud.google.com/endpoints/docs/grpc/authenticating-users#receiving_authentication_results_in_your_api) and [here](https://cloud.google.com/endpoints/docs/openapi/migrate-to-esp-v2#handle-jwt) specifies that the metadata key `X-Endpoint-API-Userinfo` key added by ESP includes `claims`:

"If available in the JWT, ESP adds the values of the id, issuer, email, and audiences properties to the encoded JSON object. It also adds the claims property that includes the original payload of the JWT. The JSON object has the form: (see below)"

```JSON
{
  "id": "from-sub",
  "issuer": "from-iss",
  "email": "from-email",
  "audiences": ["from-aud"],
  "claims": {
     original-jwt-payload
   }
}
```

ESPv2 does not appear to include `claims` and it does not reflect the documented structure which maps e.g. `sub` --> `id`, `iss` --> `issuer`.

I receive the following (pretty-printed) which is an accurately (!) decoded version of the Firebase JWT plus e.g. `firebase` provided by the service config:

```JSON
{
    "name": "Daz Wilkin",
    "iss": "https://securetoken.google.com/${FIREBASE_PROJECT}",
    "aud": "${FIREBASE_PROJECT}",
    "auth_time": 1623860883,
    "user_id": "${USER_ID}",
    "sub": "${USER_ID}",
    "iat": 1623869859,
    "exp": 1623873459,
    "email": "my@email.com",
    "email_verified": true,
    "firebase": {
        "identities":{
            "microsoft.com": ["[[REDACTED]]"],
            "email":["my@email.com"]
        },
        "sign_in_provider":"microsoft.com"
    }
}
```

In summary:

+ no `claims`
+ no remapping

**Comment by nareddyt** [comment]
From the pages you linked to, there is a little bit of text at the bottom:

> If you are using ESPv2, the format of the value in the header is different. See Migrate to Extensible Service Proxy V2 Beta for more information on the new format.

The format ESPv2 uses is documented here: https://cloud.google.com/endpoints/docs/grpc/migrate-to-esp-v2#handle-jwt

**Comment by nareddyt** [comment]
The text you quoted above

> If available in the JWT, ESP adds the values of the id, issuer, email, and audiences properties to the encoded JSON object. It also adds the claims property that includes the original payload of the JWT. The JSON object has the form: (see below)

Is for ESP (https://github.com/cloudendpoints/esp). The new structure with ESPv2 is what you care about :)

**Comment by TAOXUY** [comment]
Just to be precious on that doc.

```
In ESPv2, the X-Endpoint-API-UserInfo header only contains the Base64Url encoded payload of the original JWT, without modification.
```

We pass through what it is.

**Comment by TAOXUY** [comment]
I see. The context is distractive and confusing. I will update the doc.
